### ISSUE TITLE
Swimmer-v2 with PPO/TRPO or any other Mujoco environment with tuned hyperparameters?

### ISSUE THREAD

**Issue by hiteshisharma** [hill-a/stable-baselines#500]
I am looking for PPO/TRPO hyperparameters which work with swimmer. Right now I used the following and was able to match the plot in the paper but in the video, the agent does not seem to have learned.
- - - cg_damping
    - 0.1
  - - cg_iters
    - 15
  - - entcoeff
    - 0.0
  - - gamma
    - 0.995
  - - lam
    - 0.97
  - - max_kl
    - 0.01
  - - n_timesteps
    - 2000000.0
  - - policy
    - MlpPolicy
  - - timesteps_per_batch
    - 25000
  - - vf_iters
    - 5
  - - vf_stepsize
    - 0.001

I have changed timesteps_per_batch from 2048 to 25000 but not much difference in the performance.

Also there are no pretrained agent on Mujoco (v2 environments), so has anybody has tried training those environments with TRPO/PPO/DDPG?

**Comment by araffin** [comment]
Hello,
This is apparently a known problem (cf [CEM-RL](https://arxiv.org/abs/1810.01222) paper) where CEM performs much better than other methods.

There may be a problem in where the sensor is, as described in [POPLIN](https://arxiv.org/abs/1906.08649) paper: 
"Fixing the Swimmer Environments

We also notice that after an update in the Gym environments, the swimmer became unsolvable for
almost all algorithms. The reward threshold for solving is around 340 for the original swimmer, but
almost all algorithms, including the results shown in many published papers [34], will be stuck at the 130 reward local-minima. We note that this is due the fact that the velocity sensor is on the neck of the swimmer, making swimmer extremely prone to this performance local-minimum. We provide a fixed swimmer, which we name as Swimmer, by moving the sensor from the neck to the head. We believe this modification is necessary to test the effectiveness of the algorithms"


Anyway, to solve this issue, I would recommend you to do hyperparameter optimization using the rl zoo (it is included). Please keep us up to date if you succeed ;)

>Also there are no pretrained agent on Mujoco (v2 environments)

We do not provide pretrained agent as Mujoco is a proprietary physics engine.

**Comment by hiteshisharma** [comment]
@araffin a big thanks to you for such a quick and detailed response.
Now I am sure that I am not missing anything. 
If I happen to find working hyperparameters, I will surely update you. 
Looks like pybullet environments will be more trustworthy. :)

**Comment by kishanpb** [comment]
@hiteshisharma Were you successful in tuning hyperparameters for a Swimmer environment using PPO/TRPO agents?

**Comment by hiteshisharma** [comment]
not really....the best set was also stuck somewhere around 150.

**Comment by kishanpb** [comment]
@hiteshisharma can you update what this best set was?

**Comment by araffin** [comment]
The answer was to set a higher gamma value (at least it worked with SAC/TD3), for instance `gamma=0.9995`
Credits go to olivier sigaud, see https://rl-vs.github.io/rlvs2021/pg-pitfalls.html
